zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/user/DefaultUserCreator.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.engine.processing.user;
+
+import static java.util.Optional.*;
+
+import io.camunda.zeebe.engine.Loggers;
+import io.camunda.zeebe.engine.state.mutable.MutableProcessingState;
+import io.camunda.zeebe.protocol.Protocol;
+import io.camunda.zeebe.protocol.impl.record.value.user.UserRecord;
+import io.camunda.zeebe.protocol.record.intent.UserIntent;
+import io.camunda.zeebe.protocol.record.value.UserType;
+import io.camunda.zeebe.stream.api.ReadonlyStreamProcessorContext;
+import io.camunda.zeebe.stream.api.StreamProcessorLifecycleAware;
+import io.camunda.zeebe.stream.api.scheduling.Task;
+import io.camunda.zeebe.stream.api.scheduling.TaskResult;
+import io.camunda.zeebe.stream.api.scheduling.TaskResultBuilder;
+import org.slf4j.Logger;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+
+public final class DefaultUserCreator implements StreamProcessorLifecycleAware, Task {
+  public static final String DEFAULT_USER_USERNAME = "demo";
+  public static final String DEFAULT_USER_PASSWORD = "demo";
+  public static final String DEFAULT_USER_EMAIL = "demo@demo.com";
+  private static final Logger LOG = Loggers.PROCESS_PROCESSOR_LOGGER;
+  private final MutableProcessingState processingState;
+  private final BCryptPasswordEncoder passwordEncoder;
```

Comment:
With choice you mean why we use the `BCryptPasswordEncoder`? It's because Identity uses this to encrypt passwords before sending them to Zeebe. We have to make sure we use the same encryption as Identity for this default user, as they need to match the password when a user logs in.

So the flow is:

1. Zeebe creates default user with BCrypt encoded password
2. Zeebe exports User.CREATED event containing this encoded password
3. A user logs in in Identity
4. Identity pulls the exported user from elastic
5. Identity matches the entered password with the encoded password

Reply:
Yes, I was wondering why `BCryptPasswordEncoder ` instead of other options, thanks